### ISSUE TITLE
Unable to use template variable as json key

### ISSUE THREAD

**Issue by mrfelton** [strongloop/loopback-connector-rest#27]
The following fails with a fatal error:

```
{
      "template": {
        "method": "POST",
        "url": "https://example.com/v1/authenticate",
        "body": {
          "grant_type": "client_credentials",
          "client_id": "{!clientId}",
          "client_secret": "{!clientSecret}",
          "auth_data": {
            "{!uniqueId}": {
              "id": "third_party",
              "emailAddress": "{!email}"
            }
          }
        }
      },
      "functions": {
        "registerPerson": ["clientId", "clientSecret", "uniqueId", "email"]
      }
    }
```

> "TypeError: Cannot read property 'type' of undefined
>     at /Users/tom/workspace/example/example/src/node_modules/loopback-connector-rest/lib/rest-connector.js:58:51"

The problem is the use of `{!uniqueId}` as a json key.

**Comment by raymondfeng** [comment]
@mrfelton Haha, your use case is a bit special as the variable is the key. The REST connector doesn't support that with the current implementation. But I'm open for that.

**Comment by raymondfeng** [comment]
It should be fixed by loopback-connector-rest@1.7.0.
